test/state/state.cpp

```diff
@@ -156,6 +156,10 @@ std::variant<TransactionReceipt, std::error_code> transition(
         for (const auto& key : storage_keys)
             storage[key].access_status = EVMC_ACCESS_WARM;
     }
+    // EIP-3651: Warm COINBASE.
+    // We cannot touch it here because it may create an account in pre Spurious Dragon.
```

Comment:
Hm? The condition is >=Shanghai, so not sure how Spurious Dragon is related.

Reply:
This refers to the implementation detail that the coinbase is touched after the transaction execution (see some lines below). This gives 2 coinbase account lookups, but it looks we cannot do better because we are not allowed to create the account here for pre Spurious Dragon revisions.

Reply:
> because we are not allowed to create the account here for pre Spurious Dragon revisions.

Since the condition is >=Shanghai, this can never happen. Or the point of the comment is to explain the condition?

Reply:
I changed the comment.